liquibase-core/src/main/java/liquibase/diff/compare/core/SchemaComparator.java

```diff
@@ -105,15 +105,15 @@ private String getSchemaAfterComparison(Database accordingTo, String schemaName1
 
     private String getFinalSchemaAfterComparison(Database accordingTo, String schemaName1, String comparisonSchema1, String comparisonSchema2, String finalSchema1) {
         if (CatalogAndSchema.CatalogAndSchemaCase.ORIGINAL_CASE.equals(accordingTo.getSchemaAndCatalogCase())){
-            if ((comparisonSchema1 != null) && comparisonSchema1.equals(schemaName1)) {
+            if (comparisonSchema1.equals(schemaName1)) {
```

Comment:
comparisioSchema1 and 2 can be null, as they are provided by method getComparisonSchemaOrCatalog that can return null in some specific situations.

Reply:
Equals logic is right, but if comparisonSchema1 is null then calling comparisonSchema1.equals will throw the nullpointer (null.equals) .

Reply:
Applied back these checks too.